src/app/components/event-templates/event-template-add/event-template-add.component.ts

```diff
@@ -0,0 +1,134 @@
+import { Component } from '@angular/core';
+import { EventTemplateService } from '../../../service/event-template.service';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { multipleChoiceValidator } from '../../../shared/validator/multiple-choice.validator';
+import { multipleChoiceOptionValidator } from '../../../shared/validator/multiple-choice-option.validator';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormService } from '../../../service/form.service';
+import { catchError, of, tap } from 'rxjs';
+
+@Component({
+  selector: 'app-event-template-add',
+  templateUrl: './event-template-add.component.html',
+  styleUrls: ['./event-template-add.component.scss'],
+})
+export class EventTemplateAddComponent {
+  templateFormGroup: FormGroup;
+  isEditPage = false;
+  editedTemplateId?: number;
+  errorMessage = '';
+  isLoading = false;
+  constructor(
+    private eventTemplateService: EventTemplateService,
+    private formService: FormService,
+    private reactiveFormBuilder: FormBuilder,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {
+    this.templateFormGroup = reactiveFormBuilder.group({
+      name: ['', Validators.required],
+      betTemplates: this.reactiveFormBuilder.array([]),
+    });
+    if (this.route.snapshot.url[1]?.path === 'edit') {
+      this.isEditPage = true;
+      this.editedTemplateId = parseInt(this.route.snapshot.params['id']);
+      this.eventTemplateService
+        .getById(this.editedTemplateId)
+        .pipe(
+          tap((response) => {
+            this.formService.prepopulateEventTemplateForm(
```

Comment:
Why do you do this in a `tap` operator and not in the `subscribe` of the observable?